### PR TITLE
[Event Hubs] Assorted Minor Updates

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to:

  - Update change logs
  - Refine documentation around event handlers in the buffered producer
  - Adapt to changes in the `TestEnvironment` that altered behavior
  - Fix a corner case in the `AmqpConnectionScope` that could lead to timers not being properly disposed in some error scenarios.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md [hunk 1]
```diff
@@ -2,16 +2,28 @@
...
 
-### Breaking Changes
+Thank you to our developer community members who helped to make the Event Hubs client libraries better with their contributions to this release:
 
-### Bugs Fixed
+- Andrey Shihov _([GitHub](https://github.com/andreyshihov))_
+
+### Changes
+
+$### Features Added
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md:13]: I'll fix this before merging.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs [hunk 2]
```diff
@@ -886,17 +890,65 @@ public void Dispose()
...
+            if (link != null)
+            {
+                ActiveLinks.TryRemove(link, out activeTimer);
+
+                if (activeTimer != null)
+                {
+                    try
+                    {
+                        activeTimer?.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
+                        activeTimer?.Dispose();
```
Comment (christothes) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs:928]: Isn't activeTimer guaranteed to be non-null here?

**Review by christothes — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs:928]: Yes;  yes, it is.  That's a remnant of an earlier form .  I'll get that fixed up.

**Pull request merged**